photon-client/src/main/scala/com/linkedin/photon/ml/cli/game/GameDriver.scala

```diff
@@ -185,14 +189,17 @@ trait GameDriver extends PhotonParams {
   protected[game] def prepareFeatureMapsDefault(): Map[FeatureShardId, IndexMapLoader] = {
 
     val shardConfigs = getRequiredParam(featureShardConfigurations)
-    val allFeatureSectionKeys = shardConfigs.values.map(_.featureBags).reduce(_ ++ _)
-    val nameAndTermFeatureSetContainer = NameAndTermFeatureSetContainer.readNameAndTermFeatureSetContainerFromTextFiles(
+    val allFeatureBagKeys = shardConfigs.values.map(_.featureBags).reduce(_ ++ _)
+    val nameAndTermFeatureMap = NameAndTermFeatureMapUtils.readNameAndTermFeatureMapFromTextFiles(
       getRequiredParam(featureBagsDirectory),
-      allFeatureSectionKeys,
-      sc.hadoopConfiguration)
+      allFeatureBagKeys,
+      sc)
+      // persist [[NameAndTerm]] feature [[RDD]]'s as they may be used multiple times below
+      .map { case (k, v) => (k, v.persist()) }
```

Comment:
We should specify the persistence level explicitly. I think `StorageLevel.MEMORY_ONLY_SER` would be best for this use case, but I leave it to your discretion.